### DISCUSSION TITLE
How to filter postgres array that contains an element ?

### DISCUSSION THREAD

**Discussion by leo91000** [SeaQL/sea-orm#1190] [Q&A] [answered]
Hi,

I am using postgres array in my sea orm schema as `Vec<String>`.

I want to filter all rows with arrays containing a specific element.

How can I do so ?

I've already tried `Column::MyColumn.contains(...)` however it does not work on postgres : `operator does not exist: text[] ~~ text`

**Comment by billy1624** [comment, marked as answer]
Hey @leo91000, try using the `ANY(...)` function.

```rs
fruit::Entity::find()
    .filter(PgFunc::any(Expr::col(fruit::Column::CakeId)).eq( ... ))
```

The resulting SQL would be

```sql
...
WHERE ANY(cake_id) = ...
```

**Reply by phostann** [reply]
Hey guy, your answer doesn't work for me with Postgres 15.1. the database will return an error  `syntax error at or near "ANY"`. I make some changes to your answer, and now it works.
the code:
```rust
fruit::Entity::find().filter(Expr::eq(
                Expr::val(cake_id),
                Expr::expr(PgFunc::any(Expr::col(templates::Column::CakeId))),
            ))
```

**Reply by billy1624** [reply]
Thanks for the updates!! @phostann
